### PR TITLE
small changes throughout release notes

### PR DESCRIPTION
**Pull request by SethTisue**
typos, wording changes, formatting changes, some reordering towards the top

review by @adriaanm

if merged, I'll take care of propagating the changes to the blog post and the GitHub milestone

this is a lot easier to review with `git show --word-diff`

### CODE REVIEW

File: hand-written.md [hunk 1]
```diff
@@ -340,15 +365,18 @@ The [SAM conversion](http://www.scala-lang.org/files/archive/spec/2.12/06-expres
     val sam1: MySam = () => 2 // Uses SAM conversion, not the implicit
     sam1.i()                  // Returns 2
 
-To retain the old behavior, you may compile under `-Xsource:2.11`, use an explicit call to the conversion method, or disqualify the type from being a SAM (e.g. by adding a second abstract method).
+To retain the old behavior, your choices are:
+
+* compile under `-Xsource:2.11`
+* use an explicit call to the conversion method
+* disqualify the type from being a SAM (e.g. by adding a second abstract method).
```
Comment (lrytz) [inline hand-written.md]: need to check if this will work with scala-lang's markdown parser.. the rest of the file uses a different style for bullets.

**Review by lrytz — APPROVED** [review]
LGTM!

File: hand-written.md [hunk 2]
```diff
@@ -1,93 +1,120 @@
 We are very happy to announce the availability of Scala 2.12.0!
 
+## Headline features
+
 The Scala 2.12 compiler has been completely overhauled to make use of the new VM features available in Java 8:
 
   - A trait [compiles directly to an interface](#trait-compiles-to-an-interface) with default methods. This improves binary compatibility and Java interoperability.
```
Comment (adriaanm) [inline hand-written.md]: This link will need to be updated due to change to header below.

Comment (adriaanm) [reply hand-written.md]: #traits-compile-to-interfaces

**Review by adriaanm — DISMISSED** [review]
Very very nice! Needs small fix for broken link.

**Comment by adriaanm** [comment]
While you're at it, could you add a reference to the removal of the rewrite from `withFilter -> filter` done https://github.com/scala/scala/pull/5252? (https://gitter.im/scala/contributors?at=58209308df5ae9664527672a)

**Pull request merged**